src/CMakeLists.txt

```diff
@@ -218,24 +218,37 @@ if(MSVC AND HPX_WITH_SWAP_CONTEXT_EMULATION)
+# Exported targets
+################################################################################
 set(_library_types "STATIC_LIBRARY;MODULE_LIBRARY;SHARED_LIBRARY;OBJECT_LIBRARY;INTERFACE_LIBRARY")
 set(_is_executable "$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>")
 set(_is_library "$<IN_LIST:$<TARGET_PROPERTY:TYPE>,${_library_types}>")
 
-# The main exported target
+add_library(hpx_no_wrap_main INTERFACE)
 add_library(hpx INTERFACE)
+
+target_link_libraries(hpx_no_wrap_main INTERFACE hpx_some_name)
 target_link_libraries(hpx INTERFACE hpx_some_name)
 
-# hpx_interface contains additional interface options to be passed to dependent
-# targets. We create this as a separate target to easily filter out the
-# generator expressions that can't be handled by the pkgconfig file generation.
+# hpx_interface and hpx_interface_no_wrap_main contain additional interface
+# options to be passed to dependent targets. We create these as separate
+# targets to easily filter out the generator expressions that can't be handled
+# by the pkgconfig file generation.
+add_library(hpx_interface_no_wrap_main INTERFACE)
+target_link_libraries(hpx_interface_no_wrap_main INTERFACE $<${_is_executable}:$<TARGET_NAME_IF_EXISTS:hpx_init>>)
+target_link_libraries(hpx_interface_no_wrap_main INTERFACE $<${_is_executable}:$<TARGET_NAME_IF_EXISTS:HPXInternal::hpx_init>>)
+target_compile_definitions(hpx_interface_no_wrap_main INTERFACE "$<${_is_executable}:HPX_APPLICATION_NAME_DEFAULT=$<TARGET_PROPERTY:NAME>>")
+target_compile_definitions(hpx_interface_no_wrap_main INTERFACE "$<${_is_executable}:HPX_PREFIX_DEFAULT=\"${HPX_PREFIX}\">")
+target_compile_definitions(hpx_interface_no_wrap_main INTERFACE "$<${_is_executable}:HPX_APPLICATION_EXPORTS>")
+target_compile_definitions(hpx_interface_no_wrap_main INTERFACE "$<${_is_library}:HPX_LIBRARY_EXPORTS>")
+
 add_library(hpx_interface INTERFACE)
-target_link_libraries(hpx_interface INTERFACE $<${_is_executable}:$<TARGET_NAME_IF_EXISTS:hpx_init>>)
-target_link_libraries(hpx_interface INTERFACE $<${_is_executable}:$<TARGET_NAME_IF_EXISTS:HPXInternal::hpx_init>>)
-target_compile_definitions(hpx_interface INTERFACE "$<${_is_executable}:HPX_APPLICATION_NAME_DEFAULT=$<TARGET_PROPERTY:NAME>>")
-target_compile_definitions(hpx_interface INTERFACE "$<${_is_executable}:HPX_PREFIX_DEFAULT=\"${HPX_PREFIX}\">")
-target_compile_definitions(hpx_interface INTERFACE "$<${_is_executable}:HPX_APPLICATION_EXPORTS>")
-target_compile_definitions(hpx_interface INTERFACE "$<${_is_library}:HPX_LIBRARY_EXPORTS>")
+target_link_libraries(hpx_interface INTERFACE hpx_interface_no_wrap_main)
+target_link_libraries(hpx_interface INTERFACE $<${_is_executable}:$<TARGET_NAME_IF_EXISTS:hpx_wrap>>)
+target_link_libraries(hpx_interface INTERFACE $<${_is_executable}:$<TARGET_NAME_IF_EXISTS:HPXInternal::hpx_wrap>>)
+
+target_link_libraries(hpx_no_wrap_main INTERFACE hpx_interface_no_wrap_main)
```

Comment:
Add `hpx_no_wrap_main` to `hpx_targets` otherwise it doesn't get exported

Reply:
Good catch!